lib/helpers.bash

```diff
@@ -92,762 +96,741 @@ alias reload_completion="$(_make_reload_alias completion completion)"
-  else
-    TARGET=$current_tag
-    echo "Version type: stable"
-    echo "Current tag: $current_tag"
-    echo "Tag information: $BASH_IT_GIT_URL/releases/tag/$current_tag"
-  fi
-
-  echo "Compare to latest: $BASH_IT_GIT_URL/compare/$TARGET...master"
-
-  cd - &> /dev/null || return
-}
+	_param '1: Which branch to checkout to'
+	_param '2: Which type of version we are using'
+	if git checkout "$1" &> /dev/null; then
+		echo "Bash-it successfully updated."
+		echo ""
+		echo "Migrating your installation to the latest $2 version now..."
+		_bash-it-migrate
+		echo ""
+		echo "All done, enjoy!"
+		# Don't forget to restore the original pwd (called from `_bash-it-update-`)!
+		popd > /dev/null || return
+		_bash-it-restart
+	else
+		echo "Error updating Bash-it, please, check if your Bash-it installation folder (${BASH_IT}) is clean."
+	fi
+}
+
+function _bash-it-update-() {
+	_about 'updates Bash-it'
+	_param '1: What kind of update to do (stable|dev)'
+	_group 'lib'
+
+	declare silent
+	for word in "$@"; do
+		if [[ "${word}" == "--silent" || "${word}" == "-s" ]]; then
+			silent=true
+		fi
+	done
 
-_bash-it-doctor() {
-  _about 'reloads a profile file with a BASH_IT_LOG_LEVEL set'
-  _param '1: BASH_IT_LOG_LEVEL argument: "errors" "warnings" "all"'
-  _group 'lib'
+	pushd "${BASH_IT?}" > /dev/null || return
 
-  BASH_IT_LOG_LEVEL=$1
-  _bash-it-reload
-  unset BASH_IT_LOG_LEVEL
+	DIFF=$(git diff --name-status)
+	if [[ -n "$DIFF" ]]; then
+		echo -e "Local changes detected in bash-it directory. Clean '$BASH_IT' directory to proceed.\n$DIFF"
+		return 1
+	fi
+
+	if [[ -z "$BASH_IT_REMOTE" ]]; then
+		BASH_IT_REMOTE="origin"
+	fi
+
+	git fetch "$BASH_IT_REMOTE" --tags &> /dev/null
+
+	if [[ -z "$BASH_IT_DEVELOPMENT_BRANCH" ]]; then
+		BASH_IT_DEVELOPMENT_BRANCH="master"
+	fi
+	# Defaults to stable update
+	if [[ -z "$1" || "$1" == "stable" ]]; then
+		version="stable"
+		TARGET=$(git describe --tags "$(git rev-list --tags --max-count=1)" 2> /dev/null)
+
+		if [[ -z "$TARGET" ]]; then
+			echo "Can not find tags, so can not update to latest stable version..."
+			popd > /dev/null || return
+			return
+		fi
+	else
+		version="dev"
+		TARGET="${BASH_IT_REMOTE}/${BASH_IT_DEVELOPMENT_BRANCH}"
+	fi
+
+	declare revision
+	revision="HEAD..${TARGET}"
+	declare status
+	status="$(git rev-list "${revision}" 2> /dev/null)"
+	declare revert
+
+	if [[ -z "${status}" && "${version}" == "stable" ]]; then
+		revision="${TARGET}..HEAD"
+		status="$(git rev-list "${revision}" 2> /dev/null)"
+		revert=true
+	fi
+
+	if [[ -n "${status}" ]]; then
+		if [[ -n "${revert}" ]]; then
+			echo "Your version is a more recent development version ($(git log -1 --format=%h HEAD))"
+			echo "You can continue in order to revert and update to the latest stable version"
+			echo ""
+			log_color="%Cred"
+		fi
+
+		for i in $(git rev-list --merges --first-parent "${revision}"); do
+			num_of_lines=$(git log -1 --format=%B "$i" | awk 'NF' | wc -l)
```

Comment:
I think one less external program can be used here.

```suggestion
			num_of_lines=$(git log -1 --format=%B "$i" | awk '!/^[[:space:]]*$ {++i} END{print i}')
```

Reply:
I just realized this could have been a more simple `awk 'NF {++i} END{print i}'`, idk if you prefer it. This is just to not count one trailing newline anyways, so, sorry for nitpicking on something insignificant :grin: